cmake/onnxruntime_unittests.cmake

```diff
@@ -1268,6 +1268,7 @@ if (NOT onnxruntime_ENABLE_TRAINING_TORCH_INTEROP)
               LIBS ${onnxruntime_shared_lib_test_LIBS}
               DEPENDS ${all_dependencies}
       )
+      onnxruntime_add_include_to_target(onnxruntime_api_tests_without_env onnx Boost::mp11)
```

Comment:
Actually I don't know what this test is for. if it is for testing our public APIs for the shared library, it should not link to onnx and should not use internal headers because normally a process should not have duplicated symbols. Though there are ways to deal with duplications, most people would not want to jump into the mess. 
Maybe we should delete this test instead.

Reply:
Not sure of the history but I don't think there's much value to it. 

https://github.com/microsoft/onnxruntime/blob/6949cfaf94f97e93463d77b304a3a5cdb19aa1de/cmake/onnxruntime_unittests.cmake#L1263

Not clear why we need a separate test for things that don't require the env being created first, given the general approach when using ORT is to always create the env as the first step. Additionally, it's potentially confusing as only parts of SessionOptions do not require the env to be created. If you're registering an EP in the session options that requires a logger if there's a failure, which requires the env.

@pranavsharma any reason to keep this around?

Reply:
It was done to fix an old regression. Can be safely deleted now.